### ISSUE TITLE
1.12.2 - Filter is ignored when importing items into a Refined Storage network

### ISSUE THREAD

**Issue by kosakriszi** [SleepyTrousers/EnderIO#4904]
#### Issue Description:

At the end of this [this issue](https://github.com/SleepyTrousers/EnderIO/issues/4887) it was requested that a separate issue be opened for this problem, this was not done.

Anyway, here's the deal: I have a conduit importing items into my RS system, with an advanced filter filtering out golden swords ignoring damage level. Swords are being imported despite this filter.

#### What happens:

Items are imported regardless of being blacklisted in the filter.

#### What you expected to happen:

The swords are not imported and remain in the inventory.

#### Steps to reproduce:

1. Hook up an RS conduit to an inventory with a damaged item.
2. Place in a filter that blacklists that damaged item ignoring damage.
3. Observe that this in fact does a whole lot of nothing.

EDIT: removed screenshot of my filter cause it showed whitelist instead of blacklist. That was dumb of me and it actually was an issue on blacklist.
____
#### Affected Versions (Do *not* use "latest"):

- EnderIO: **5.0.33**
- EnderCore: **1.12.2-0.5.39**
- Minecraft: **1.12.2**
- Forge: **14.23.4.2759**
- SpongeForge? **no**
- Optifine? **no**
- Single Player and/or Server? **server**

**Comment by EpicSquid** [comment]
Thanks for making the new issue. Though it hasn't been officially fixed, can you confirm this is the case with the latest ender io and Ender core?

**Comment by kosakriszi** [comment]
To my pleasant surprise this seems to be fixed on the latest version of EnderIO.

I tested with:
EnderIO: **5.0.36**
EnderCore: **1.12.2-0.5.41**
Minecraft: **1.12.2**
Forge: **14.23.5.2770**

Now to just wait for the modpack I play to update. Sorry for the trouble!

**Comment by aranian** [comment]
This is also a problem when exporting items from the network. I cannot get any export filter to respect the settings for damage, NBT and Metadata. Only exact matches are ever exported.

**Comment by ogradyjd** [comment]
I'm going to add myself to this list.  I'm having all the problems as described by this issue, and also issues #5053 and #5076. 
 
# TL;DR
This is happening in the [All The Mods - Remix](https://minecraft.curseforge.com/projects/all-the-mods-3-remix) pack.
```
Minecraft Version: 1.12.2
(modded) forge-14.23.5.2815
Ender IO                 CrazyPants_MC     EnderIO-1.12.2-5.0.40.jar
Ender IO Endergy         tterrag1098       EnderIO-endergy-1.12.2-5.0.40.jar
Ender Storage 1.8.+      covers1624        EnderStorage-1.12.2-2.4.5.135-universal.jar
EnderCore                tterrag1098       EnderCore-1.12.2-0.5.45.jar
```
# Issues

Import from machines and storage (Actually Additions crates or vanilla chests) ignores filters (basic item filter, big item filter) on whitelisting.  Items are imported regardless of the whitelist.
Behaviour is sporadic.  Sometimes import works. Sometimes it does not.  I have not yet figured out what the magic series of actions is that enables/disables expected behaviour.

One thing I can add that I don't think others have seen is that when you have more  than one conduit attached to the side of a machine, sometimes switching between conduit tabs, the filter boxes from the Refined Storage Conduit UI ghost on to the, in my case, ender energy conduit UI.

![2019-04-06_10-11-33](https://user-images.githubusercontent.com/4014049/55670591-6cf4f200-5854-11e9-927f-555ec3824855.png)

# Info
Updated: 3/25/2019
Version: 1.2.0
Minecraft Version: 1.12.2
(modded) forge-14.23.5.2815

[ATM-R_ModList.txt](https://github.com/SleepyTrousers/EnderIO/files/3050555/ATM-R_ModList.txt)

I can zip up my world and send it wherever you want if you want it to look at.  Let me know.

**Comment by marikc0** [comment]
RS conduits also are not working correctly for me. I'm able to import into a TE magma crucible and export from a TE Igneous Extruder, both using filters, but I can't import or export with Industrial Foregoing machines whatsoever either by using filters or without filters. No matter what I change with them I can't move items.

**Comment by HenryLoenwind** [comment]
Please note that export filtering is severely limited by RS's filtering capabilities. It will not export a wrong item, but it may get stuck when RS wants to export a wrong item and our filter stops it. As long as RS doesn't decide to export another item, nothing will happen.
